xsd-core/src/main/java/io/ballerina/xsd/core/visitor/XSDVisitorImpl.java

```diff
@@ -0,0 +1,613 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.xsd.core.visitor;
+
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.xsd.core.Utils;
+import io.ballerina.xsd.core.XSDFactory;
+import io.ballerina.xsd.core.component.ComplexType;
+import io.ballerina.xsd.core.component.Element;
+import io.ballerina.xsd.core.component.XSDComponent;
+import io.ballerina.xsd.core.component.SimpleType;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.xsd.core.visitor.VisitorUtils.MAX_OCCURS;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.MIN_OCCURS;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.addNamespace;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.asIterable;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.deriveType;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.generateDefaultValue;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.handleDefaultValues;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.handleFixedValues;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.handleMaxOccurrences;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.handleMinOccurrences;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.isSimpleType;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.convertToCamelCase;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.sanitizeString;
+import static io.ballerina.xsd.core.visitor.VisitorUtils.typeGenerator;
+
+/**
+ * This class is responsible for visiting and processing components of an XSD schema.
+ *
+ * @since 0.1.0
+ */
+public class XSDVisitorImpl implements XSDVisitor {
+    public static final String PUBLIC = "public";
+    public static final String WHITESPACE = " ";
+    public static final String CONTENT_FIELD = "\\#content";
+    public static final String SEMICOLON = ";";
+    public static final String RECORD = "record";
+    public static final String TYPE = "type";
+    public static final String NAME = "name";
+    public static final String DEFAULT = "default";
```

Comment:
We can define this in a new file called Constants. It will reduce the same constant declare in two files

Reply:
I haven't used a separate class for constants as it is considered as an anti pattern